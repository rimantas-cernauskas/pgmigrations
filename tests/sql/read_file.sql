-- Test function pgmigrations.read_file
\i tests/sql/setup/init.sql

/*

This script relieas on the existence of the PG_VERSION file

*/

BEGIN;
SELECT plan(3);

SET ROLE __pgmigrations_testuser;

PREPARE my_query AS
    SELECT pgmigrations.read_file( 'PG_VERSION' );

SELECT throws_ok(
    'my_query'
    ,'must be superuser to call read_file()'
    ,'read_file() should throw exception when called as unprivileged user'
);

DEALLOCATE my_query;

SET ROLE __pgmigrations_superuser;

SELECT matches(
     pgmigrations.read_file( 'PG_VERSION')
    ,'\d+'
    ,'read_file() should return file content when called as super user'
);

SELECT is(
     pgmigrations.read_file( 'should_not_exist')
    ,NULL
    ,'read_file() should return NULL when file does not exist'
);

SELECT * FROM finish();
ROLLBACK;