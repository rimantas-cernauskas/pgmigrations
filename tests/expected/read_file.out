-- Test function pgmigrations.read_file
\i tests/sql/setup/init.sql
\set ECHO errors
1..3
ok 1 - read_file() should throw exception when called as unprivileged user
ok 2 - read_file() should return file content when called as super user
ok 3 - read_file() should return NULL when file does not exist
